# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1456820714 18000
#      Tue Mar 01 03:25:14 2016 -0500
# Node ID eb7287db3da9a3d8b723ea72c347e5ba4eafbd4f
# Parent  e5d96f39a37d23410942c573414ec3ab874d082e
create .oct and other files in build tree (bug #46260)

* Makefile.am (ALL_LOCAL_TARGETS): Add $(OCTAVE_INTERPRETER_FILES) to
the list.
(all-local): Simplify by adding all dependencies to ALL_LOCAL_TARGETS.
* libinterp/module.mk: (ALL_LOCAL_TARGETS): Delete.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -170,17 +170,24 @@ MAINTAINERCLEANFILES += \
   ChangeLog \
   $(BUILT_DISTFILES)
 
 CONFIG_FILES = @ac_config_headers@ @ac_config_files@
 
 octinclude_HEADERS += oct-conf-post.h
 nodist_octinclude_HEADERS += config.h octave-config.h
 
-all-local: $(ALL_LOCAL_TARGETS) $(noinst_SCRIPTS) $(DIST_INFO_FILES) .gdbinit $(DOC_TARGETS)
+ALL_LOCAL_TARGETS += \
+  $(OCTAVE_INTERPRETER_TARGETS) \
+  $(noinst_SCRIPTS) \
+  $(DIST_INFO_FILES) \
+  .gdbinit \
+  $(DOC_TARGETS)
+
+all-local: $(ALL_LOCAL_TARGETS)
 	@echo ""
 	@echo "Octave successfully built.  Now choose from the following:"
 	@echo ""
 	@echo "   ./run-octave    - to run in place to test before installing"
 	@echo "   make check      - to run the tests"
 	@echo "   make install    - to install (PREFIX=$(prefix))"
 	@echo ""
 
diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -141,18 +141,16 @@ include libinterp/dldfcn/module.mk
 if AMCOND_ENABLE_DYNAMIC_LINKING
   OCT_FILES = $(DLDFCN_LIBS:.la=.oct)
   DLD_LIBOCTINTERP_LIBADD = libinterp/liboctinterp.la
 else
   OCT_FILES =
   DLD_LIBOCTINTERP_LIBADD =
 endif
 
-ALL_LOCAL_TARGETS += $(OCT_FILES)
-
 libinterp_liboctinterp_la_SOURCES = \
   libinterp/octave.cc \
   libinterp/version.cc \
   $(LIBINTERP_OPERATORS_SRC) \
   $(TEMPLATE_INST_SRC)
 
 nodist_libinterp_liboctinterp_la_SOURCES = \
   libinterp/corefcn/mxarray.h \
